src/publish.js

```diff
@@ -440,10 +445,11 @@ have published it.\
         if (rename?.length > 0) { originalName = pack.name; }
 
         let firstTimePublishing;
+        let tag;
```

Comment:
Along with the two changes you mention in your review summary, this change is crucial. It moves `tag` to the outer scope so that it can be consumed later on.